amulety/amulety.py

```diff
@@ -133,55 +245,353 @@ def embed_airr(
+        if chain not in supported_chains:
+            raise ValueError(
+                f"Model 'balm-paired' requires paired chains (HL/LH). Use --chain HL or --chain LH instead of --chain {chain}."
+            )
+        if chain == "LH":
+            warnings.warn(
+                "Model 'balm-paired' was trained with H-L order. Using L-H order may reduce accuracy.",
+                UserWarning,
+            )
+
+        # Process data for BALM-paired - use concat mode for paired chains
+        X, dat = process_chain_data(
+            airr,
+            chain,
+            internal_chain,
+            sequence_col,
+            cell_id_col,
+            duplicate_col,
+            receptor_type,
+            model_type="paired_only",
+        )
+
         embedding = balm_paired(sequences=X, cache_dir=cache_dir, batch_size=batch_size)
+
     # TCR models
     elif model == "tcr-bert":
+        # Check compatible chains
+        supported_chains = ["H", "L", "HL", "LH", "H+L"]
+        if chain not in supported_chains:
+            raise ValueError(f"Model tcr-bert only accepts {', '.join(supported_chains)} inputs! Got: {chain}")
+
+        # TCR-BERT: Only CDR3, supports H+L, H, L, HL/LH
+        X, dat = process_chain_data(
+            airr, chain, internal_chain, sequence_col, cell_id_col, duplicate_col, receptor_type, model_type="standard"
+        )
+
         embedding = tcr_bert(sequences=X, cache_dir=cache_dir, batch_size=batch_size)
+
+    elif model == "tcremp":
+        # TCREMP requires CDR3 sequences - check if we have CDR3 data available
+        from amulety.utils import get_cdr3_sequence_column
+
+        # Check if this is TCR data and if CDR3 columns are available
+        tcr_loci = {"TRA", "TRB", "TRG", "TRD"}
+        present_loci = set(airr["locus"].unique())
+        is_tcr_data = bool(present_loci & tcr_loci)
+
+        if is_tcr_data:
+            # For TCR data, try to get the best CDR3 column
+            effective_sequence_col = get_cdr3_sequence_column(airr, sequence_col)
+            if effective_sequence_col == sequence_col and sequence_col not in ["junction_aa", "cdr3_aa"]:
+                logger.warning(
+                    f"TCREMP was trained on CDR3 sequences, but using {sequence_col}. Consider using junction_aa or cdr3_aa columns if available."
+                )
+        else:
+            # For non-TCR data, require explicit CDR3 column
+            if sequence_col not in ["CDR3_aa", "junction_aa", "cdr3_aa"]:
+                raise ValueError(
+                    "Model tcremp was trained on CDR3 aa sequences. For non-TCR data, please provide CDR3_aa, junction_aa, or cdr3_aa as sequence column."
+                )
+        # Check compatible chains
+        supported_chains = ["H", "L", "HL", "LH", "H+L"]
+        if chain not in supported_chains:
+            raise ValueError(f"Model tcremp only accepts {', '.join(supported_chains)} inputs! Got: {chain}")
```

Comment:
```suggestion
            warnings.warn(f"Model tcremp was trained on {', '.join(supported_chains)}. These chains are recommended instead of : {chain}.")
```